### PR TITLE
basic account forms with semantic ui #10 #11

### PR DESCRIPTION
**Pull request by zaklampert**
***Not merge ready.***

This pr replaces `accounts-ui` with `useracounts:core` and `useracounts:unstyled` for custom registration, signin, and forgot password form styles. We're simply following patterns from [meteor/todos](http://github.com/meteor/todos) here.

Todo: 
1) Routing with JustinRouter conventions and determining the usefulness of `AccountsTemplates.configureRoute` 
2) Reset-Password flow (email)
3) Move to own package (?)
4) Clean up markup and add other semantic-ui components
5) Loading States

### CODE REVIEW

File: imports/ui/layouts/header.js [hunk 1]
```diff
@@ -9,4 +12,13 @@ const HEADER_HELPERS = {
   routes: () => _.map(Router.routeMap, route => route),
 };
 
+const HEADER_EVENTS = {
+  'click #logout'() {
+    Meteor.logout();
+    // QUESTION: Does JustinRouter have a push function? (zl)
+    FlowRouter.go('login');
```
Comment (justinr1234) [inline imports/ui/layouts/header.js]: `FlowRouter.go(Router.routeMap.APP_LOGIN.path)`

File: imports/ui/layouts/header.html [hunk 2]
```diff
@@ -6,8 +6,11 @@
         <a class="{{ isActivePath exampleAddPage }} item" href="{{ exampleAddPage }}">Add Example</a>
         <a class="{{ isActivePath exampleListPage }} item" href="{{ exampleListPage }}">List Example</a>
         {{> DEBUG_ROUTE_MENU routes=routes }}
+        <a href="#" class="right aligned item" id="logout">Logout</a>
+        {{else}}
```
Comment (justinr1234) [inline imports/ui/layouts/header.html]: Indent

**Comment by justinr1234** [comment]
Thanks Zak. I would recommend using a prebuilt package as there's a lot of them out there. Overriding directly using aldeed's override package is a pretty good idea, but I would think using something like this (or the many others that exist) would save time and be more vetted: https://github.com/meteor-useraccounts/semantic-ui/

Additionally you can configure the routes using `Router.routeMap.TEMPLATE_NAME_HERE.path` overall.

File: imports/ui/layouts/header.js [hunk 3]
```diff
@@ -1,12 +1,21 @@
...
 const HEADER_HELPERS = {
   homePage: () => Router.routeMap.HOME_PAGE.path,
   exampleAddPage: () => Router.routeMap.EXAMPLE_ADD.path,
   exampleListPage: () => Router.routeMap.EXAMPLE_LIST.path,
   routes: () => _.map(Router.routeMap, route => route),
 };
 
+const HEADER_EVENTS = {
+  'click #logout'() {
+    Meteor.logout();
```
Comment (justinr1234) [inline imports/ui/layouts/header.js]: While not exactly bad, you should follow the patterns of the library:

https://github.com/meteor-useraccounts/semantic-ui/blob/master/lib/at_nav_button.js#L2

https://github.com/meteor-useraccounts/core/blob/9dd207acd5545cd85242c3dd0eb74b32c40b48c9/lib/templates_helpers/at_nav_button.js#L12

But likely you can just use the `atNavButton` template here.

File: imports/api/helpers/both/routes/routes.js [hunk 4]
```diff
@@ -16,9 +16,7 @@ export const routes = {
   },
   APP_LOGIN: {
     path: '/login',
-  },
-  APP_REGISTER: {
-    path: '/register',
+    action: () => defaultBlazeRender('AUTH_PAGE'),
```
Comment (justinr1234) [inline imports/api/helpers/both/routes/routes.js]: Please add new routes as here: https://github.com/justinr1234/opinionated-meteor-application-template/blob/master/packages/justinr1234-example/imports/routes.js

File: imports/ui/layouts/header.html [hunk 5]
```diff
@@ -6,8 +6,11 @@
         <a class="{{ isActivePath exampleAddPage }} item" href="{{ exampleAddPage }}">Add Example</a>
         <a class="{{ isActivePath exampleListPage }} item" href="{{ exampleListPage }}">List Example</a>
         {{> DEBUG_ROUTE_MENU routes=routes }}
+        <a href="" class="right aligned item" id="logout">Logout</a>
+      {{else}}
+        <a href="/login" class="right aligned item" >Login</a>
       {{/if}}
-      <a href="#" class="right aligned item">{{> loginButtons align="right"}} </a>
```
Comment (justinr1234) [inline imports/ui/layouts/header.html]: Should be able to use this I think: https://github.com/meteor-useraccounts/semantic-ui/blob/master/lib/at_nav_button.html

File: package.json [hunk 6]
```diff
@@ -2,8 +2,8 @@
   "name": "opinionated-meteor-starter",
   "private": false,
   "scripts": {
-    "start": "meteor --no-release-check --settings ./config/settings.development.json",
-    "start-dev": "DEBUG=* meteor --no-release-check --settings ./config/settings.development.json",
+    "start": "meteor --no-release-check --settings ./config/settings.development.json ",
+    "start-dev": "DEBUG=* meteor --no-release-check --settings ./config/settings.development.json --port 8008",
```
Comment (justinr1234) [inline package.json]: Why the change to port 8008?

Comment (zaklampert) [inline package.json]: My bad. Didn't mean to commit that change...I've got about 6 other meteor apps running.

Comment (justinr1234) [inline package.json]: No problem, I understand!

File: packages/justinr1234-example/imports/routes.js [hunk 7]
```diff
@@ -19,6 +19,10 @@ export const routes = {
   EXAMPLE_EDIT: {
     path: '/example/:_id',
   },
+  APP_LOGIN: {
+    path: '/login',
+    action: () => defaultBlazeRender('AUTH_PAGE'),
```
Comment (justinr1234) [inline packages/justinr1234-example/imports/routes.js]: I'm not sure there's a reason to call it `AUTH_PAGE` here, so you could just make the route and template called `APP_LOGIN` preferably. Unless I'm missing something? Very well could be! :)

**Comment by zaklampert** [comment]
@justinr1234 there doesn't seem to be a good way to use your opinionated routing that I can see and use the pre-fabbed account templates. In the past i've just custom rolled account management from scratch because I wasn't happy with the [recommended approach to using the flowrouting library](https://guide.meteor.com/accounts.html#useraccounts).

**Comment by justinr1234** [comment]
@zaklampert Can you elaborate a little more what issues you're referring to specifically? I'm curious to know what you ran into! Thanks!
